### PR TITLE
[Feature]: Termux Support

### LINKED ISSUE #1201: [Feature]: Termux Support
## Android Termux Support
Problem: Can't install lunarvim in termux

### Issues:
  - Termux doesn't have `sudo`
  - `cp: cannot create file '/usr/local/bin' : No such file or directory`

### Solution
  - Detect termux with the ".termux/" directory (Maybe)
  - Not using `sudo`
  - Using `$HOME/../usr/bin` or `$HOME/.local/bin`(not sure if this one will work) instead of `/usr/local/bin`

### What worked on my phone
  - Removed `sudo`
  - Changed `/usr/local/bin` to `$HOME/../usr/bin`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by rynpix**
# Description

Adds Termux support for lunarvim

Fixes #1201 

## How Has This Been Tested?

Tested on my phone & wsl ubuntu 20.04 LTS

Test Commands: 
  - `LVBRANCH=rolling bash <(curl -s https://raw.githubusercontent.com/rynpix/lunarvim/rolling/utils/installer/uninstall.sh)`
  - `LVBRANCH=rolling bash <(curl -s https://raw.githubusercontent.com/rynpix/lunarvim/rolling/utils/installer/install.sh)`

#### NOTE: I had some issues while installing I was getting:
```
module 'toggleterm' not found:^Ino field package.preload['toggleterm']^Ino file './toggleterm.lua'^Ino file '/usr/share/luajit-2.1.0-beta3/toggleterm.lua'^Ino file '/usr/local/share/lua/5.1/toggleterm.lua'^Ino file '/usr/local/share/lua/5.1/toggleterm/init.lua'^Ino file '/usr/share/lua/5.1/toggleterm.lua'^Ino file '/usr/share/lua/5.1/toggleterm/init.lua'^Ino file '/home/rayane/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/toggleterm.lua'^Ino file '/home/rayane/.cache/nvim/packer_hererocks/2.1.0-beta3/share/lua/5.1/toggleterm/init.lua'^Ino file '/home/rayane/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/toggleterm.lua'^Ino file '/home/rayane/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/luarocks/rocks-5.1/toggleterm/init.lua'^Ino file './toggleterm.so'^Ino file '/usr/local/lib/lua/5.1/toggleterm.so'^Ino file '/usr/lib/x86_64-linux-gnu/lua/5.1/toggleterm.so'^Ino file '/usr/local/lib/lua/5.1/loadall.so'^Ino file '/home/rayane/.cache/nvim/packer_hererocks/2.1.0-beta3/lib/lua/5.1/toggleterm.so'Error running config for nvim-lspinstall
Error detected while processing VimEnter Autocommands for "*":
E5108: Error executing lua ...m/site/pack/packer/start/packer.nvim/lua/packer/load.lua:115: Vim(echomsg):E114: Missing quote: "Error in packer_compiled: ...m/site/pack/packer/start/packer.nvim/lua/packer/load.lua:52: [string \"...\"]:0: module 'lspinstall' not found:
Press ENTER or type command to continue
```
and I had to `^Z` and restart the installation with  the `--overwrite` flag
#### NOTE2: I had bad times trying to fix the indent with lunarVim
- Check logs
- ...

### CODE REVIEW

File: utils/installer/install.sh [hunk 1]
```diff
@@ -63,6 +69,10 @@ installpiponubuntu() {
 	sudo apt install python3-pip >/dev/null
 }
 
+installpipontermux() {
+	apt install python
```
Comment (abzcoding) [inline utils/installer/install.sh:73]: will this install `pip` as well?

Comment (rynpix) [reply utils/installer/install.sh:73]: Termux doesn't come with python3 by default to install it we use `apt install python` and it installs `pip`, make, clang ...
btw i just retested it and it works

**Comment by rebuilt** [comment]
I tested on my phone.  This is my first use of termux.  I installed f-droid.  Then used f-droid it install termux.  Is this the right termux to install or is there another version?  This termux uses arch but not pacman.  I installed packages with `pkg install python`.  I had to install python, npm, and git.  I cant install any language servers.   I  don't own a bluetooth keyboard and im getting duplicated  keyboard entries so it's very slow to test everything.  If i type `ls` it registers `lsls`  so I don't think I'll be able to make more progress testing this without a bluetooth keyboard

**Comment by rebuilt** [comment]
I didn't encounter the toggleterm issue.  All plugins installed ok

**Comment by rebuilt** [comment]
I also never used the the overwrite flag

**Comment by rebuilt** [comment]
Testing the windows install is still on my todo list.

**Comment by abzcoding** [comment]
It didn't work on my phone either : 🤔

**Comment by rebuilt** [comment]
> It didn't work on my phone either thinking

```
pkg install neovim
pkg install python
pkg install npm
pkg install git
```

And even after installing, you don't get any of the 'goodies' from lunarvim.  No formatting, no linting, no language servers.  Compe text-sourced tab completion didn't work.   It would be better to use standard neovim on a phone.

**Comment by rynpix** [comment]
Screenshot_2021-08-02-13-16-43-956_com termux

**Comment by rebuilt** [comment]
I couldn't install stylua or sumneko.  Do those work for you?  What steps are we missing to install this correctly?

**Comment by rynpix** [comment]
how to i install theme?

**Comment by rebuilt** [comment]
> how to i install theme?

No not a theme.  How did you install a language server? 

Screenshot_20210802-144156_Termux

**Comment by rebuilt** [comment]
Are you using a bluetooth keyboard?  I don't get snippet or text completion either.  I'm wondering what the difference is.

**Comment by rynpix** [comment]
no im not using external keyboard

**Comment by rynpix** [comment]
yes lua lsp doesn't work
Screenshot_2021-08-02-13-49-00-892_com termux

**Comment by rebuilt** [comment]
@abzcoding If we put this in, it should be `use at your own risk`.  I'm going to be able to answer questions about how to set this up properly are you?

**Comment by abzcoding** [comment]
> @abzcoding If we put this in, it should be `use at your own risk`. I'm going to be able to answer questions about how to set this up properly are you?

I have no idea how to use it properly tbh :sweat_smile: let me test some stuff, brb

**Comment by rynpix** [comment]
i tried calling the lsp from outside lvim
```
~ $ node .local/share/nvim/lspinstall/lua/sumneko-lua-language-server 
/data/data/com.termux/files/home/.local/share/nvim/lspinstall/lua/sumneko-lua-language-server:2
$(dirname $0)/sumneko-lua/extension/server/bin/Linux/lua-language-server -E -e LANG=en $(dirname $0)/sumneko-lua/extension/server/main.lua $*
  ^^^^^^^

SyntaxError: missing ) after argument list
    at wrapSafe (internal/modules/cjs/loader.js:979:16)
    at Module._compile (internal/modules/cjs/loader.js:1027:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

**Comment by abzcoding** [comment]
Yeah it works good enough I guess

File: utils/installer/install.sh [hunk 2]
```diff
@@ -100,6 +110,14 @@ installpacker() {
 }
 
 cloneconfig() {
+	if [ -d "/data/data/com.termux" ]; then
+		sudo() {
+			eval "$@"
+		}
+		bin_path="$HOME/../usr/bin"
```
Comment (abzcoding) [inline utils/installer/install.sh:117]: can you define this variable outside of the function
like this for example
```shell
declare USER_BIN_DIR="/usr/local/bin"
```
and then use and change it wherever you want

File: utils/installer/uninstall.sh [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+	sudo() {
+		eval "$@"
+	}
+	bin_path="$HOME/../usr/bin"
+else
+	bin_path='/usr/local/bin'
+fi
 rm -rf ~/.local/share/lunarvim
-sudo rm /usr/local/bin/lvim
+sudo rm $bin_path/lvim
```
Comment (abzcoding) [inline utils/installer/uninstall.sh:11]: also this

File: utils/installer/install.sh [hunk 4]
```diff
@@ -110,13 +110,12 @@ installpacker() {
 }
 
 cloneconfig() {
+	declare USER_BIN_DIR="/usr/local/bin"
```
Comment (abzcoding) [inline utils/installer/install.sh]: please move this to line #1 of the file 😅

Comment (rynpix) [reply utils/installer/install.sh]: I moved it to line 6

**Comment by abzcoding** [comment]
🤦 my bad, but can you fix these 🥺 
just remove `declare` and use double quote everywhere
```
declare USER_BIN_DIR="/usr/local/bin"
^------------------^ SC2039: In POSIX sh, 'declare' is undefined.


In ./utils/installer/uninstall.sh line 10:
sudo rm $USER_BIN_DIR/lvim
        ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
sudo rm "$USER_BIN_DIR"/lvim


In ./utils/installer/install.sh line 6:
declare USER_BIN_DIR="/usr/local/bin"
^------------------^ SC2039: In POSIX sh, 'declare' is undefined.


In ./utils/installer/install.sh line 132:
	sudo cp "$HOME/.local/share/lunarvim/lvim/utils/bin/lvim" $USER_BIN_DIR
                                                                  ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
	sudo cp "$HOME/.local/share/lunarvim/lvim/utils/bin/lvim" "$USER_BIN_DIR"


In ./utils/installer/install.sh line 133:
	sudo chmod a+rx $USER_BIN_DIR/lvim
                        ^-----------^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
	sudo chmod a+rx "$USER_BIN_DIR"/lvim
```

**Comment by rynpix** [comment]
do i move it to line 2 or 4

**Comment by abzcoding** [comment]
> do i move it to line 2 or 4

current position of that is fine, just make it look like this
```shell
USER_BIN_DIR="/usr/local/bin"
```
instead of
```shell
declare USER_BIN_DIR="/usr/local/bin"
```

File: utils/installer/install.sh (hunk 4 above)
Comment (rynpix) [reply utils/installer/install.sh]: @abzcoding I moved it into line 6 
Should I move it to 2 or 4?

Comment (abzcoding) [reply utils/installer/install.sh]: line 4 would be much appreciated 🥺

**Comment by abzcoding** [comment]
double quotes are still an issue 😢

**Pull request merged**